src/Alevin.cpp

```diff
@@ -549,9 +549,15 @@ bool writeFastq(AlevinOpts<ProtocolT>& aopt,
       for (size_t i = 0; i < rangeSize; ++i) { // For all the read in this batch
         auto& rp = rg[i];
         if(aopt.protocol.end == bcEnd::FIVE){
-          barcode = rp.first.seq.substr(0, aopt.protocol.barcodeLength);
-          umi = rp.first.seq.substr(aopt.protocol.barcodeLength,
-                                    aopt.protocol.umiLength);
+          //barcode = rp.first.seq.substr(0, aopt.protocol.barcodeLength);
+          //umi = rp.first.seq.substr(aopt.protocol.barcodeLength,
+          //                          aopt.protocol.umiLength);
+          // TODO: The above isn't valid for CEL-Seq2. I temporarily commented 
+          //       it out and replaced with a CEL-Seq2-specific format so that 
+          //       I could test my changes.
+          barcode = rp.first.seq.substr(aopt.protocol.umiLength, 
```

Comment:
Nice catch, I missed updating this part. You are right we should call the extract barcode and umi functions here.